google-cloud-spanner/src/test/java/com/google/cloud/spanner/connection/SpannerPoolTest.java

```diff
@@ -145,6 +151,9 @@ public void testGetSpanner() {
     spanner1 = pool.getSpanner(options3, connection1);
     spanner2 = pool.getSpanner(options4, connection2);
     assertThat(spanner1).isNotEqualTo(spanner2);
+    spanner1 = pool.getSpanner(options7, connection1);
+    spanner2 = pool.getSpanner(options8, connection2);
+    assertNotEquals(spanner1, spanner2);
```

Comment:
Prefer `assertThat` method over `assertNotEquals` method. It provides readability and type check benefits and is also consistent with what is used in existing test.